From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shlomi Marco <s.marco@rubycomm.com>
Date: Thu, 27 Feb 2026 00:00:00 +0000
Subject: arm64: dts: rockchip: Enable PCIe on NanoPi Zero2

Enable the PCIe Gen2x1 controller and combo PHY on the FriendlyElec
NanoPi Zero2 to support the M.2 Key-E 2230 slot used for WiFi cards.

The reset GPIO (gpio1 RK_PA2) and pinctrl (pciem1_pins) match the
vendor kernel device tree (rk3528-nanopi-rev01.dts).

Signed-off-by: Shlomi Marco <s.marco@rubycomm.com>
---
 arch/arm64/boot/dts/rockchip/rk3528-nanopi-zero2.dts | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3528-nanopi-zero2.dts b/arch/arm64/boot/dts/rockchip/rk3528-nanopi-zero2.dts
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/rockchip/rk3528-nanopi-zero2.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3528-nanopi-zero2.dts
@@ -176,6 +176,17 @@
 	cpu-supply = <&vdd_arm>;
 };

+&combphy {
+	status = "okay";
+};
+
+&pcie {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pciem1_pins>;
+	reset-gpios = <&gpio1 RK_PA2 GPIO_ACTIVE_HIGH>;
+	status = "okay";
+};
+
 &gmac1 {
 	clock_in_out = "output";
 	phy-handle = <&rgmii_phy>;
--
Armbian

